Datamapping with streamlit CLI update - Standalone version (#1202)
Snowflake-Labs/sfquickstarts: site/sfguides/src/data_mapping_in_native_apps/data_mapping_in_native_apps.md

The first solution is not really what we want to be doing because the consumer will potentially have to create objects on their Snowflake instance just to satisfy the application's requirements, so this Quickstart will deliver the second solution.

> aside positive
> 
> **Note** - The following steps explain how to create the app using the Snowflake CLI, as this option presents as a faster, more straightforward way to build Native Apps. You can also accomplish it by manually creating the folder structure and executing the SQL commands in Snowsight, but this alternative is out of the scope for this Quickstart.

<!-- ------------------------ -->
## Project Structure
Duration: 7

The project we are about to create is composed of several files in the following tree structure:

```
-- Datamapping App project

    |-- app
    |   |-- setup_script.sql
    |   |-- manifest.yml
    |   |-- ui
    |   |   |-- enricher_dash.py
    |-- scripts
    |   |-- setup-package-script.sql
    |-- snowflake.yml
    |-- prepare_data.sh
    
```

> To start creating this folder structure, go to your console and inside the folder you want the project to reside. Execute the following **`snow app init datamapping_app_streamlit`**. This will create several files for a basic template, we are replacing some of them and also creating new ones.

Reviewer: ```suggestion
To start creating this folder structure, go to your console and inside the folder you want the project to reside. Execute the following command: **`snow app init datamapping_app_streamlit`**. 

This will create several files from our basic template; we will replacing some of them, as well as create some new ones.
```

GRANT USAGE ON SCHEMA ENRICHIP TO APPLICATION ROLE APP_PUBLIC;

--this is an application version of the object shared with the application package
CREATE VIEW IF NOT EXISTS ENRICHIP.LITEDB11 AS SELECT * FROM IP2LOCATION.litedb11;

> -- If the user prefers, access can be granted directly to the IPLOCATION.litebd11 view, created in the setup_package_script file.

Reviewer: Because this is going in a versioned schema I suggest `CREATE OR REPLACE` as a best practice. The only reason it might be necessary is when an ALTER APPLICATION UPGRADE or CREATE APPLICATION fails, in which case the next attempt at installation will use the same versioned schema "containers". We likely want the versions from the more-recent attempt to overwrite any objects created from earlier instances; correctness here is more important than making sure the subsequent setup is as fast as possible.

--accepts an IP address and returns a modified version of the IP address 
--the modified version will be used in the lookup

> CREATE SECURE FUNCTION IF NOT EXISTS ENRICHIP.ip2long(ip_address varchar(16))

Reviewer: Same as above

--converts it using the ip2long function above
--looks up the returned value in the view
--returns the enhanced information as an object

> CREATE SECURE FUNCTION IF NOT EXISTS ENRICHIP.ip2data(ip_address varchar(16))

Reviewer: Same as above

--this is the reference to the streamlit (not the streamlit itself)
--this was referenced in the manifest file

> create streamlit if not exists ui."Dashboard" from 'ui' main_file='enricher_dash.py';

Reviewer: Same as above
